trieste/acquisition/function.py

```diff
@@ -1443,37 +1461,54 @@ def prepare_acquisition_function(
         :param model: The model over the specified ``dataset``.
         :param pending_points: The points we penalize with respect to.
         :return: The (log) expected improvement penalized with respect to the pending points.
-        :raise tf.errors.InvalidArgumentError: If the first call does not have pending_points=None,
-            or ``dataset`` is empty.
+        :raise tf.errors.InvalidArgumentError: If the ``dataset`` is empty.
         """
         tf.debugging.assert_positive(len(dataset))
-        tf.debugging.Assert(pending_points is None, [])
 
-        # no penalization required if no pending_points.
-        return self._update_base_acquisition_function(dataset, model)
+        acq = self._update_base_acquisition_function(dataset, model)
+        if pending_points is not None:
```

Comment:
Not sure whether this change is actually necessary, as the very first time pending_points should always be None right? Still, if we do make the change then we should probably also handle passing in empty pending_points.

```suggestion
        if pending_points is not None or len(pending_points) == 0:
```

Reply:
I guess that's an attempt at being general. Consider some complex ask/tell scenarios where optimizer instance is created anew for each ask. I agree that's rather hypothetical, but it seems easy to support, hence this change. Will cover the empty tensor case, thanks.